libs/datamodel/core/src/transform/ast_to_dml/validate.rs

```diff
@@ -95,61 +94,129 @@ impl<'a> Validator<'a> {
     pub(crate) fn post_standardisation_validate(
         &self,
         ast_schema: &ast::SchemaAst,
-        schema: &dml::Datamodel,
+        datamodel: &dml::Datamodel,
         diagnostics: &mut Diagnostics,
     ) {
-        let constraint_names = NamesValidator::new(schema, self.source);
+        let mut diagnostics_2 = self.validate_constraint_names_connector_specific(ast_schema, datamodel);
 
-        for model in schema.models() {
-            let ast_model = ast_schema.find_model(&model.name).expect(STATE_ERROR);
+        diagnostics.append(&mut diagnostics_2);
+        for model in datamodel.models() {
+            let mut new_errors = self.validate_relation_arguments_bla(
+                datamodel,
+                ast_schema.find_model(&model.name).expect(STATE_ERROR),
+                model,
+            );
 
-            if let Some(pk) = &model.primary_key {
-                if let Some(name) = &pk.db_name {
-                    // Only for SQL Server for now...
-                    if constraint_names.is_duplicate(name) {
-                        let span = ast_model.id_attribute().span;
+            diagnostics.append(&mut new_errors);
+        }
+    }
 
-                        let message = format!(
-                            "The given constraint name `{}` is already in use in the data model. Please provide a different name using the `map` argument.",
-                            name
+    fn validate_constraint_names_connector_specific(
+        &self,
+        ast_schema: &ast::SchemaAst,
+        datamodel: &dml::Datamodel,
+    ) -> Diagnostics {
+        let mut diagnostics = Diagnostics::new();
+
+        if let Some(source) = self.source {
```

Comment:
We could reduce nesting by one level here with an

```rust
let source = if let Some(source) = self.source { source } else { return diagnostics };
```

Reply:
Done.